File: generator/app/templates/app/src/main/java/com/example/base/baseapp/MyApplication.java
package com.example.base.baseapp;


import com.example.base.baseapp.dagger.AppComponent;
import com.example.base.baseapp.dagger.module.AppModule;

public class MyApplication extends BaseApplication<AppComponent> {

    @Override
    protected AppComponent initAppComponent() {
        return DaggerAppComponent
                .builder()
                .appModule(new AppModule(this))
                .build();
    }
}

File: generators/app/templates/app/src/main/kotlin/com/example/base/baseapp/MyApplication.kt
package com.example.base.baseapp;


import com.example.base.baseapp.dagger.AppComponent;
import com.example.base.baseapp.dagger.module.AppModule;
import com.example.base.baseapp.dagger.DaggerAppComponent;

class MyApplication : BaseApplication<AppComponent>() {

    override fun initAppComponent(): AppComponent {
        return DaggerAppComponent
                .builder()
                .appModule(AppModule(this))
                .build()
    }
}
